### PR TITLE
VMBroker conversion rex in upgrades

### PR DESCRIPTION
**Pull request by pondrejk**
Needs review from upgrade specialists :) Also I wasn't sure how to do the vm cleanup postupgrade

### CODE REVIEW

File: tests/upgrades/test_remoteexecution.py [hunk 1]
```diff
@@ -96,17 +95,7 @@ def test_pre_scenario_remoteexecution_external_capsule(self, request, default_lo
...
-        )
-        client.create()
-        request.addfinalizer(
-            lambda: cleanup_of_provisioned_server(
-                hostname=client.hostname,
-                provisioning_server=self.libvirt_vm,
-                distro=DISTRO_RHEL7,
-            )
-        )
+        client = VMBroker(nick='rhel7', host_classes={'host': ContentHost})
```
Comment (jyejare) [inline tests/upgrades/test_remoteexecution.py:98]: Instead, you should use the existing function level fixture :
https://github.com/SatelliteQE/robottelo/blob/master/pytest_fixtures/broker.py#L47

File: tests/upgrades/test_remoteexecution.py [hunk 2]
```diff
@@ -207,17 +193,8 @@ def test_pre_scenario_remoteexecution_satellite(
...
-        client.create()
-        request.addfinalizer(
-            lambda: cleanup_of_provisioned_server(
-                hostname=client.hostname,
-                provisioning_server=self.libvirt_vm,
-                distro=DISTRO_RHEL7,
-            )
-        )
+
+        client = VMBroker(nick='rhel7', host_classes={'host': ContentHost})
```
Comment (jyejare) [inline tests/upgrades/test_remoteexecution.py:197]: Instead, you should use the existing function level fixture :
https://github.com/SatelliteQE/robottelo/blob/master/pytest_fixtures/broker.py#L47

**Review by jyejare — CHANGES_REQUESTED** [review]
ACK pending comments.

File: tests/upgrades/test_remoteexecution.py [hunk 3]
```diff
@@ -96,17 +95,7 @@ def test_pre_scenario_remoteexecution_external_capsule(self, request, default_lo
...
-        )
-        client.create()
-        request.addfinalizer(
-            lambda: cleanup_of_provisioned_server(
-                hostname=client.hostname,
-                provisioning_server=self.libvirt_vm,
-                distro=DISTRO_RHEL7,
-            )
-        )
+        client = rhel7_contenthost
```
Comment (jyejare) [inline tests/upgrades/test_remoteexecution.py:98]: You can directly use it, you need not initialise the `client` variable for it.

Below line will change to:
```
rhel7_contenthost.install_capsule_katello_ca(capsule=self.proxy_name)
```
Same others.

**Comment by jyejare** [comment]
trigger: test-robottelo
pytest: -m pre_upgrade tests/upgrades/test_remoteexecution.py

**Review by mshriver — APPROVED** [review]
The test failures in PRT are well outside the scope of what you're doing here.

Your conversion to using VMBroker content hosts is clean.

**Pull request merged**